### PR TITLE
fix web service example

### PR DESCRIPTION
**Pull request by glorysdj**
this PR is to fix model can not loading problems in web service sample
also add a unified style to train and inference with a same processing method defined in scala.

---
This change is [Reviewable](https://reviewable.io/reviews/intel-analytics/analytics-zoo/680)

### CODE REVIEW

File: apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/training/TextClassificationTrainer.scala [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        (text, labelId)
+      })
+    }).flatten
+    val classNum = categoryPathList.length
+    log.info(s"Found ${data.length} texts.")
+    log.info(s"Found ${classNum} classes")
+    (data, classNum)
+  }
+
+  def buildModel(embeddingFile: File, wordToIndexMap: MutableMap[String, Int], sequenceLength: Int, classNum: Int, encoderOutputDim: Int = 256)(implicit ev: TensorNumeric[Float]) = {
```
Comment (hkvision) [inline apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/training/TextClassificationTrainer.scala]: Seems you can use a TextClassifier model directly?

File: apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/training/TextClassificationTrainer.scala [hunk 2]
```diff
@@ -0,0 +1,151 @@
+package com.intel.analytics.zoo.apps.textclassfication.training
+
+import java.io.File
```
Comment (hkvision) [inline apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/training/TextClassificationTrainer.scala]: Wondering why do you need to include train code in web service? As most of the code is almost the same as the Scala example.

Comment (glorysdj) [reply apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/training/TextClassificationTrainer.scala]: In previous version of web service it relies on existing textclassifer model, but when the textclassifer changed, web service failed to load the model. In this version, we define a simple example model to decouple with the textclassifer, so both of them can be changed without affect each other.
A more important reason is that it simulates a real use case, customer will define a model and set up a series of pre and post  processing in scala, train it and save the model, then load it in pure java env or web service, and leverage the existing processing in scala to do the the inference.

**Comment by hkvision** [comment]
Is the training accuracy compatible with the original example?

File: apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/training/TextClassificationTrainer.scala (hunk 1 above)
Comment (hkvision) [inline apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/training/TextClassificationTrainer.scala]: is ev used?

File: apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/training/TextClassificationTrainer.scala [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+      val trainDataDirFile = new File(params.trainDataDir)
+      require(trainDataDirFile.exists(), "train data dir dose not exist")
+      val embeddingFile = new File(params.embeddingFile)
+      require(embeddingFile.exists(), "glove data file dose not exist")
+      val stopWordsCount = 10
+      val sequenceLength = 200
+      val partitionNum = params.partitionNum
+      val batchSize = params.batchSize
+      val nbEpoch = params.nbEpoch
+      val modelSeveDirPath = params.modelSeveDirPath
```
Comment (hkvision) [inline apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/training/TextClassificationTrainer.scala]: typo save..

File: apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/training/TextClassificationTrainer.scala [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    }
+
+    parser.parse(args, TextClassificationTrainerParams()).map(params => {
+      println(params)
+      val trainDataDirFile = new File(params.trainDataDir)
+      require(trainDataDirFile.exists(), "train data dir dose not exist")
+      val embeddingFile = new File(params.embeddingFile)
+      require(embeddingFile.exists(), "glove data file dose not exist")
+      val stopWordsCount = 10
+      val sequenceLength = 200
```
Comment (hkvision) [inline apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/training/TextClassificationTrainer.scala]: For sequenceLength=200, users must input the glove 200d, shall we specify this?

File: apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    }
+  }
+
+  def doStopWords(tokens: List[String], stopWords: List[String]): List[String] = {
+    tokens.filter(!stopWords.contains(_))
+  }
+
+  def doStopWords(tokens: List[String], stopWordsCount: Int): List[String] = {
+    val topK = ListMap(tokens.groupBy(identity).mapValues(_.length).toSeq.sortBy(-_._2): _*).take(stopWordsCount)
+    tokens.filter(!topK.keySet.contains(_))
```
Comment (hkvision) [inline apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala]: Can this call the above doStopWords directly?

File: apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  def doShaping(tokens: List[String], sequenceLength: Int, truncateFromLeft: Boolean = true): List[String] = {
+    tokens.length > sequenceLength match {
+      case true => {
+        truncateFromLeft match {
+          case true => tokens.take(sequenceLength)
```
Comment (hkvision) [inline apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala]: truncateFromLeft should mean keeping the right part of the sequence? Is the true/false actions reversed?

File: apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    val shapedTokens = doShaping(stoppedTokens, sequenceLength)
+    val indexes = doIndex(shapedTokens, wordToIndexdMap)
+    Tensor(indexes.map(_.asInstanceOf[Float]).toArray, Array(sequenceLength))
+  }
+
+  def doPreprocess(text: String, stopWordsCount: Int, sequenceLength: Int, embeddingFile: File): Tensor[Float] = {
+    val tokens = doTokenize(text)
+    val stoppedTokens = doStopWords(tokens, stopWordsCount)
+    val shapedTokens = doShaping(stoppedTokens, sequenceLength)
+    val wordToIndexdMap = doLoadWordToIndexMap(embeddingFile)
```
Comment (hkvision) [inline apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala]: Call the above method after getting the map? Most of the code is duplicate

File: apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+import scala.io.Source
+
+trait TextProcessing {
+
+  val logger = LoggerFactory.getLogger(getClass)
+
+  def doTokenize(text: String, toLowerCase: Boolean = true): List[String] = {
+    val replaceed = text.replaceAll("[^a-zA-Z]", " ")
+    toLowerCase match {
+      case true => replaceed.toLowerCase().split("\\s+").filter(_.size > 2).toList
```
Comment (hkvision) [inline apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala]: Shall we put toLowerCase() separately as the remaining split/filter for both cases are the same

File: apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/training/TextClassificationTrainer.scala (hunk 1 above)
Comment (glorysdj) [reply apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/training/TextClassificationTrainer.scala]: yes, it is used in a implicit way.
model.add(Convolution1D(encoderOutputDim, 5, activation = "relu")) need it

File: apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/training/TextClassificationTrainer.scala (hunk 3 above)
Comment (glorysdj) [reply apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/training/TextClassificationTrainer.scala]: fixed

File: apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala (hunk 5 above)
Comment (glorysdj) [reply apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala]: yes, but need to transfer topK to List, it is not necessary imo.

File: apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala (hunk 6 above)
Comment (glorysdj) [reply apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala]: fixed

File: apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala (hunk 7 above)
Comment (glorysdj) [reply apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala]: fixed

File: apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+  def doStopWords(tokens: List[String], stopWordsCount: Int): List[String] = {
+    val topK = ListMap(tokens.groupBy(identity).mapValues(_.length).toSeq.sortBy(-_._2): _*).take(stopWordsCount)
+    tokens.filter(!topK.keySet.contains(_))
+  }
+
+  def doShaping(tokens: List[String], sequenceLength: Int, truncateFromRigtht: Boolean = true): List[String] = {
+    tokens.length > sequenceLength match {
+      case true => {
+        truncateFromRigtht match {
```
Comment (hkvision) [inline apps/web-service-sample/text-classification-training/src/main/scala/com/intel/analytics/zoo/apps/textclassfication/processing/TextProcessing.scala]: typo...

**Pull request merged**